pkg/promtail/targets/filetargetmanager.go

```diff
@@ -120,6 +120,12 @@ func NewFileTargetManager(
 		}
 		tm.syncers[cfg.JobName] = s
 		config[cfg.JobName] = cfg.ServiceDiscoveryConfig
+
+		// Add Source value to the static config target groups for unique identification
+		// within scrape pool.
+		for i, tg := range cfg.ServiceDiscoveryConfig.StaticConfigs {
+			tg.Source = fmt.Sprintf("%d", i)
+		}
```

Comment:
Could this be moved before `targetSyncer` is initialized? I understand that this will work as-is (given that `StaticConfigs` is a pointer) but I'd prefer to treat it as a dependency of configuring the `targetSyncer`.

Reply:
Made the changes

Reply:
We can even add this here https://github.com/grafana/loki/blob/master/pkg/promtail/targets/manager.go#L54

Reply:
Yeah, I had looked into other places to add it but I think where you have it now makes the most sense - it's the only place in the code where we use SD configs so it feels tidy to keep it close to where it's being used.

Reply:
Cool! Thanks